File: Demo Storyboard/LGSideMenuControllerDemo/TableViewController.m
//
//  TableViewController.m
//  LGSideMenuControllerDemo
//
//  Created by Grigory Lutkov on 05.11.15.
//  Copyright (c) 2015 Grigory Lutkov. All rights reserved.
//

#import "TableViewController.h"
#import "MainViewController.h"
#import "NavigationController.h"
#import "ViewController.h"

@interface TableViewController ()

@property (strong, nonatomic) NSArray *titlesArray;

@end

@implementation TableViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    _titlesArray = @[@"Style Scale From Big",
                     @"Style Slide Above",
                     @"Style Slide Below",
                     @"Style Scale From Little",
                     @"Landscape always visible",
                     @"Status bar always visible",
                     @"Status bar light content",
                     @"Custom style"];
}

#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _titlesArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    cell.textLabel.font = [UIFont systemFontOfSize:16.f];
    cell.textLabel.text = _titlesArray[indexPath.row];
    return cell;
}

#pragma mark - UITableViewDelegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];

    UINavigationController *navigationController = [storyboard instantiateViewControllerWithIdentifier:@"NavigationController"];
    MainViewController *mainViewController = [storyboard instantiateInitialViewController];
    mainViewController.rootViewController = navigationController;

    if (indexPath.row == 0)
        [mainViewController setupWithPresentationStyle:LGSideMenuPresentationStyleScaleFromBig type:0];
    else if (indexPath.row == 1)
        [mainViewController setupWithPresentationStyle:LGSideMenuPresentationStyleSlideAbove type:0];
    else if (indexPath.row == 2)
        [mainViewController setupWithPresentationStyle:LGSideMenuPresentationStyleSlideBelow type:0];
    else if (indexPath.row == 3)
        [mainViewController setupWithPresentationStyle:LGSideMenuPresentationStyleScaleFromLittle type:0];
    else if (indexPath.row == 4)
        [mainViewController setupWithPresentationStyle:LGSideMenuPresentationStyleScaleFromBig type:1];
    else if (indexPath.row == 5)
        [mainViewController setupWithPresentationStyle:LGSideMenuPresentationStyleSlideAbove type:2];
    else if (indexPath.row == 6)
        [mainViewController setupWithPresentationStyle:LGSideMenuPresentationStyleSlideAbove type:3];
    else if (indexPath.row == 7)
        [mainViewController setupWithPresentationStyle:0 type:4];

    UIWindow *window = [UIApplication sharedApplication].delegate.window;

    window.rootViewController = mainViewController;

    [UIView transitionWithView:window
                      duration:0.3
                       options:UIViewAnimationOptionTransitionCrossDissolve
                    animations:nil
                    completion:nil];
}

@end

File: Demo_Swift_Storyboard/LGSideMenuControllerDemo/TableViewController.swift
//
//  TableViewController.swift
//  LGSideMenuControllerDemo
//
//  Created by Grigory Lutkov on 05.11.15.
//  Copyright © 2015 Grigory Lutkov <Friend.LGA@gmail.com>. All rights reserved.
//

import UIKit

class TableViewController: UITableViewController {

    private let titlesArray = ["Style Scale From Big",
                               "Style Slide Above",
                               "Style Slide Below",
                               "Style Scale From Little",
                               "Landscape always visible",
                               "Status bar always visible",
                               "Status bar light content",
                               "Custom style"]

    // MARK: - UITableViewDataSource
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return titlesArray.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)

        cell.textLabel!.font = UIFont.systemFont(ofSize: 16.0)
        cell.textLabel!.text = titlesArray[indexPath.row]
        
        return cell
    }
    
    // MARK: - UITableViewDelegate
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let storyboard = UIStoryboard(name: "Main", bundle: nil)
        
        let navigationController = storyboard.instantiateViewController(withIdentifier: "NavigationController") as! UINavigationController
        let mainViewController = storyboard.instantiateInitialViewController() as! MainViewController
        mainViewController.rootViewController = navigationController

        switch indexPath.row {
        case 0:
            mainViewController.setup(presentationStyle: .scaleFromBig, type: 0)
            break
        case 1:
            mainViewController.setup(presentationStyle: .slideAbove, type: 0)
            break
        case 2:
            mainViewController.setup(presentationStyle: .slideBelow, type: 0)
            break
        case 3:
            mainViewController.setup(presentationStyle: .scaleFromLittle, type: 0)
            break
        case 4:
            mainViewController.setup(presentationStyle: .scaleFromBig, type: 1)
            break
        case 5:
            mainViewController.setup(presentationStyle: .slideAbove, type: 2)
            break
        case 6:
            mainViewController.setup(presentationStyle: .slideAbove, type: 3)
            break
        case 7:
            mainViewController.setup(presentationStyle: .slideAbove, type: 4)
            break
        default:
            break
        }

        let window = UIApplication.shared.delegate!.window!!
        window.rootViewController = mainViewController

        UIView.transition(with: window, duration: 0.3, options: [.transitionCrossDissolve], animations: nil, completion: nil)
    }

}
